cpp/tests/groupby/group_nth_element_test.cpp

```diff
@@ -252,6 +252,51 @@ TYPED_TEST(groupby_nth_element_test, exclude_nulls)
     test_single_agg(keys, vals, expect_keys, expect_vals2, std::move(agg));
 }
 
+TYPED_TEST(groupby_nth_element_test, exclude_nulls_negative_index)
+{
+    using K = int32_t;
+    using V = TypeParam;
+    using R = experimental::detail::target_type_t<V, experimental::aggregation::NTH_ELEMENT>;
+
+    fixed_width_column_wrapper<K> keys({ 1, 2, 3, 3, 1, 2, 2, 1, 3, 3, 2, 4, 4, 2},
+                                       { 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1});
+    fixed_width_column_wrapper<V> vals({ 0, 1, 2, 2, 3, 4, 5, 6, 7, 8, 9, 4, 4, 2},
+                                       { 0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0});
+
+    //  { 1, 1, 1     2, 2, 2,        3, 3,    4}
+    fixed_width_column_wrapper<K> expect_keys({ 1,          2,              3,       4}, all_valid());
+    //  { -, 3, 6     1, 4, -, 9, -   2, 2, 8,   4,-}
+    //0  null,        value,          value,     value
+    //1  value,       value,          value,     null
+    //2  value,       null,           value,     out
+    //3  out,         value,          out,       out
+    //4  out,         null,           out,       out
+
+    //include_nulls::YES
+    fixed_width_column_wrapper<R> expect_nuls0{{6,         -1,              8,       -1}, {1,0,1,0}};
+    fixed_width_column_wrapper<R> expect_nuls1{{3,          9,              2,        4}};
+    fixed_width_column_wrapper<R> expect_nuls2{{-1,        -1,              2,       -1}, {0,0,1,0}};
+    
+    //include_nulls::NO
+    fixed_width_column_wrapper<R> expect_vals0{{6,          9,              8,        4}};
+    fixed_width_column_wrapper<R> expect_vals1{{3,          4,              2,       -1}, {1,1,1,0}};
+    fixed_width_column_wrapper<R> expect_vals2{{-1,         1,              2,       -1}, {0,1,1,0}};
+
+    auto 
```

Comment:
This `auto` here kind of bugs me. I'd prefer `std::unique_ptr<aggregation> agg;`

Alternatively, `agg` could be eliminated entirely:
```cpp
test_single_agg(keys, vals, expect_keys, expect_nuls0,
  cudf::experimental::make_nth_element_aggregation(-1, cudf::include_nulls::YES));
```

Reply:
AFAIK, Usage of `auto` is usually encouraged.
why do you not prefer `auto`?

Reply:
I put the `agg = ...` because, it kind of serves like self explaining comment before the actual test code line comes below. It's easy to read.

Reply:
`auto` is good, I just thought it was strange for it to be on a line by itself. But, I suppose `clang-format` is going to have the final say in formatting now anyway! I'll just resolve this comment

Reply:
`clang-format` put `auto` in single line.